pkg/collector/resource.go

```diff
@@ -145,9 +145,13 @@ func IgnoreIfKustomizationFound(kustomizations []ksapi.Kustomization) FilterFunc
 }
 
 // IgnoreRuleExclusions returns a FilterFunc which excludes resources part of configuration exclusions.
-func IgnoreRuleExclusions(exclusions []v1.Exclusion) FilterFunc {
+func IgnoreRuleExclusions(cluster string, exclusions []v1.Exclusion) FilterFunc {
 	return func(res unstructured.Unstructured, logger klog.Logger) bool {
 		for _, exclusion := range exclusions {
+			if !matchesCluster(cluster, exclusion.Cluster) {
+				continue
+			}
```

Comment:
I reckon it would be more clear to have the cluster and resources separated in the config. Mixing resources and cluster doesn't make it obvious. A consumer could also think it would only exclude that resource from the configured cluster which is not the case here.

I reckon it makes sense to have two different rulesets for this:
(And cluster in excludeRrsources might also be supported but in that case only for the targeted resource).
```
apiVersion: gitopszombies/v1
kind: Config
excludeClusters:
- remote-cluster-1
excludeResources:
- name: default
  apiVersion: v1
  kind: ServiceAccount
- name: velero-capi-backup-.*
  namespace: velero
  apiVersion: velero.io/v1
  kind: Backup
```

Reply:
Actually that was the purpose of it. I want to be able to apply a filter on some clusters and not on some others.
Here I use velero only on my management cluster so having `backup.velero.io/v1` on it make sense whereas having it on workload clusters is not intended.
To me, the all cluster processing exclusion can stay at command line level don't you think ? Otherwize you should make it possible to configure all command line options in this configuration file.

Reply:
Since a config is already introduced every command line flag can also also be defined in the config thats why I proposed it to be that way.
All other flags can be defined in a separate pr though.

Reply:
OK let's do it in another PR then. I plane also to add a prune option 😅

Reply:
Can we rename `exclusions` to `excludeResources` in the config api?